src/functional/iterator/embedded.py

```diff
@@ -178,12 +178,13 @@ class Column(np.lib.mixins.NDArrayOperatorsMixin):
     and simplify dispatching in iterator ir builtins.
     """
 
-    def __init__(self, kstart: int, data: np.ndarray) -> None:
+    def __init__(self, kstart: int, data: np.ndarray | Scalar) -> None:
         self.kstart = kstart
         self.data = data
 
     def __getitem__(self, i: int) -> Any:
-        result = self.data[i - self.kstart]
+        assert isinstance(self.data, (np.ndarray, Scalar))
+        result = self.data[i - self.kstart] if isinstance(self.data, np.ndarray) else self.data
```

Comment:
not sure in which state this PR is, but this seems wrong.

Reply:
yes, I should have converted it back to draft. I discussed this change yesterday with @tehrengruber and he agreed. We  can talk about ti after the standup

Reply:
I'll push a slightly different implementation, making it a full array (as other functions rely on data to be an ndarray).